### ISSUE TITLE
Wish: Report `250 2.0.0 Ok: queued as 17409e25-7190-4d09-954f-5f0ca794940f`

### ISSUE THREAD

**Issue by stappersg** [tweedegolf/mailcrab#64]
Hello,

A _nice to have_ would be `mailcrab` reporting to sending side with which reference the message was queued.  Simular as a MTA as `postfix` does.

```text
$ swaks --to stappers@stappers.it
=== Trying rosa.stappers.it:25...
=== Connected to rosa.stappers.it.
<-  220 rosa.stappers.it ESMTP Postfix (Debian/GNU)
 -> EHLO redacted
    ...
 -> 
 -> .
<-  250 2.0.0 Ok: queued as E191A1FD3
 -> QUIT
<-  221 2.0.0 Bye
=== Connection closed with remote host.
$
```

It would be something like
```diff
--- a/backend/src/mail_server.rs
+++ b/backend/src/mail_server.rs
@@ -93,13 +93,13 @@ impl mailin::Handler for MailHandler {
         Ok(())
     }
 
-    fn data_end(&mut self) -> mailin::Response {
+    fn data_end(&mut self, id: &str) -> mailin::Response {
         if let Err(e) = self.parse_mail() {
             event!(Level::WARN, "Error parsing email: {}", e);
 
             mailin::response::Response::custom(500, "Error parsing message".to_string())
         } else {
-            mailin::response::OK
+            mailin::response::Response::custom(250, "2.0.0 Ok: queued as {}".to_string(), id)
         }
     }
 
```

But that will need a change in the [`mailin`](https://gitlab.com/alienscience/mailin) library.

Maybe there are other options.

**Comment by stappersg** [comment]
> ... will need a change in the [`mailin`](https://gitlab.com/alienscience/mailin) library.

[_requested_](https://gitlab.com/alienscience/mailin/-/issues/12)

**Comment by stappersg** [comment]
With the information from https://gitlab.com/alienscience/mailin/-/issues/12#note_1417098369 I was able to make merge request #67

**Comment by stappersg** [comment]
Merge request #67 revoked because the `self.id` being used wasn't the correct one.

**Comment by marlonbaeten** [comment]
Fixed with https://github.com/tweedegolf/mailcrab/pull/68

**Comment by stappersg** [comment]
> Fixed with #68

Yeah, it works. Thanks. Meanwhile I have seen https://github.com/tweedegolf/mailcrab/commit/92b48d019336296a62485a532d62b711f95120b8 and probably have seen what was needed for getting #67 working. In other words:  I'm gratefull for the change and do think that it would have been good thing if what went into commit 92b48d01933 was split into dedicated commits.

**Comment by stappersg** [comment]
>  ... do think that it would have been good thing if what went into commit https://github.com/tweedegolf/mailcrab/commit/92b48d019336296a62485a532d62b711f95120b8 was split into dedicated commits.

That became #69 ...
